static/app/views/performance/newTraceDetails/traceMetadataHeader.tsx

```diff
@@ -20,16 +25,344 @@ interface TraceMetadataHeaderProps {
+  path: string,
+  query: Location['query'],
+  organization: Organization
+) {
+  return {
+    pathname: normalizeUrl(`/organizations/${organization.slug}/${path}`),
+    // Remove traceView specific query parameters that are not needed when navigating back.
+    query: {...omit(query, ['node', 'fov', 'timestamp', 'eventId'])},
+  };
+}
+
+function getPerformanceBreadCrumbs(organization: Organization, location: Location) {
+  const crumbs: Crumb[] = [];
+
+  crumbs.push({
+    label: t('Performance'),
+    to: getBreadCrumbTarget(`performance`, location.query, organization),
+  });
+
+  switch (location.query.tab) {
+    case Tab.EVENTS:
+      crumbs.push({
+        label: t('All Events'),
+        to: getBreadCrumbTarget(
+          `performance/summary/events`,
+          location.query,
+          organization
+        ),
+      });
+      break;
+    case Tab.TAGS:
+      crumbs.push({
+        label: t('Tags'),
+        to: getBreadCrumbTarget(`performance/summary/tags`, location.query, organization),
+      });
+      break;
+    case Tab.SPANS:
+      crumbs.push({
+        label: t('Spans'),
+        to: getBreadCrumbTarget(
+          `performance/summary/spans`,
+          location.query,
+          organization
+        ),
+      });
+
+      const {spanSlug} = location.query;
+      if (spanSlug) {
+        crumbs.push({
+          label: t('Span Summary'),
+          to: getBreadCrumbTarget(
+            `performance/summary/spans/${spanSlug}`,
+            location.query,
+            organization
+          ),
+        });
+      }
+      break;
+    case Tab.AGGREGATE_WATERFALL:
+      crumbs.push({
+        label: t('Transaction Summary'),
+        to: getBreadCrumbTarget(
+          `performance/summary/aggregateWaterfall`,
+          location.query,
+          organization
+        ),
+      });
+      break;
+    default:
+      crumbs.push({
+        label: t('Transaction Summary'),
+        to: getBreadCrumbTarget(`performance/summary`, location.query, organization),
+      });
+      break;
+  }
+
+  crumbs.push({
+    label: t('Trace View'),
+  });
+
+  return crumbs;
+}
+
+function getIssuesBreadCrumbs(organization: Organization, location: Location) {
+  const crumbs: Crumb[] = [];
+
+  crumbs.push({
+    label: t('Issues'),
+    to: getBreadCrumbTarget(`issues`, location.query, organization),
+  });
+
+  if (location.query.groupId) {
+    crumbs.push({
+      label: t('Issue Details'),
+      to: getBreadCrumbTarget(
+        `issues/${location.query.groupId}`,
+        location.query,
+        organization
+      ),
+    });
+  }
+
+  crumbs.push({
+    label: t('Trace View'),
+  });
+
+  return crumbs;
+}
+
+function getInsightsModuleBreadcrumbs(location: Location, organization: Organization) {
+  const crumbs: Crumb[] = [];
+
+  if (organization.features.includes('performance-insights')) {
+    crumbs.push({
+      label: t('Insights'),
+    });
+
+    switch (location.query.referrer) {
```

Comment:
are you sure you want to use referrer for this? referrer is often stripped by adblockers to prevent tracking and our app tends to use referrer only for click tracking

Reply:
Ahh I see, @scttcper, help me understand this a bit. Would changing `referrer` to something like `entry_point` work?

Reply:
@Abdkhan14 yeah entryPoint would make sense. or tabConfiguration. Some indication that the query parameter is controlling what is on the page. I wouldn't expect referrer to change anything

Reply:
Got it, I'll make the change.

Reply:
I think `source` fits the best here.